android/app/src/main/java/org/fossasia/openevent/viewmodels/SessionDetailActivityViewModel.java

```diff
@@ -0,0 +1,77 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+import org.fossasia.openevent.utils.ConstantStrings;
+import org.fossasia.openevent.utils.SharedPreferencesUtil;
+
+import io.realm.RealmChangeListener;
+
+public class SessionDetailActivityViewModel extends ViewModel {
+
+    public static final String BY_ID = "id";
+    public static final String BY_NAME = "name";
+
+    private RealmDataRepository realmRepo;
+    private MutableLiveData<Session> sessionByIdLiveData;
+    private MutableLiveData<Session> sessionByNameLiveData;
+    private Session sessionById;
+    private Session sessionByName;
+
+    public SessionDetailActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<Session> getSessionById(int id, String loadedFlag) {
+        if (sessionByIdLiveData == null) {
+            sessionByIdLiveData = new MutableLiveData<>();
+            sessionById = realmRepo.getSession(id);
+            sessionById.addChangeListener((RealmChangeListener<Session>) loadedSession -> {
+                if (!loadedSession.isValid())
+                    return;
+
+                if (loadedFlag == null || loadedFlag.equals(BY_ID)) {
+                    SharedPreferencesUtil.putInt(ConstantStrings.SESSION_MAP_ID, id);
```

Comment:
Don't use any Android Specific class inside ViewModel. All Android specific work should be done in fragments/activities